test/widgets/user_test.dart

```diff
@@ -78,5 +79,26 @@ void main() {
       check(await actualUrl(tester, avatarUrl)).isNull();
       debugNetworkImageHttpClientProvider = null;
     });
+    testWidgets('shows placeholder when image URL gives error', (WidgetTester tester) async {
+      addTearDown(testBinding.reset);
+      prepareBoringImageHttpClient(success: false);
+      await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+      final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+      final badUser = eg.user(avatarUrl: 'https://zulip.com/avatarinvalid.png');
+      await store.addUser(badUser);
+      await tester.pumpWidget(TestZulipApp(
+        child: PerAccountStoreWidget(
+          accountId: eg.selfAccount.id,
+          child: AvatarImage(userId: badUser.userId, size: 30))));
+      await tester.pumpAndSettle();
+      expect(
+        find.descendant(
+          of: find.byType(AvatarImage),
+          matching: find.byIcon(ZulipIcons.person),
+        ),
+        findsOneWidget,
+      );
+  debugNetworkImageHttpClientProvider = null;
+    });
```

Comment:
nits:

- Empty line before a test
- Try to adjust indentation and spacing to follow the pattern of existing tests
- Use `check` instead of `expect`; see existing tests for examples